backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/Boxing.kt

```diff
@@ -89,13 +89,13 @@ private fun ValueType.initCache(context: Context, cacheName: String,
             .setConstant(true)
     val staticData = context.llvm.staticData
     val values = (start..end).map { staticData.createInitializer(kotlinType, createConstant(it)) }
-    val llvmBoxType = structType(context.llvm.runtime.objHeaderType, this.llvmType)
+    val llvmBoxType = structType(context.llvm.runtime.objHeaderType, getLLVMType(this, true))
```

Comment:
Maybe instead of parameter just different getters? `this.llvmType` and `this.llvmStorageType`?